### PR TITLE
doc: boards: add CW310 readme

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

Add OT CW310 readme.

Found using `./tools/check-for-readmes.sh` tool.

### Testing Strategy

n/a

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by alistair23 — DISMISSED** [review]

**Review by twilfredo — APPROVED** [review]

File: boards/opentitan/earlgrey-cw310/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+OpenTitan ChipWhisper CW310 Board
```
Comment (jrvanwhy) [inline boards/opentitan/earlgrey-cw310/README.md:1]: This reads as if OpenTitan is the producer of the CW310, which is not correct. Something like "OpenTitan on the ChipWhisperer CW310 Board" would be a little clearer.

**Comment by phil-levis** [comment]
@alistair23 If this is the primary board you're doing the OT work on, it needs much more of a README than this. The top-level OT README is good; perhaps just add a link saying instructions on using the board can be found there?

File: boards/opentitan/earlgrey-cw310/README.md (hunk 1 above)
Comment (phil-levis) [reply boards/opentitan/earlgrey-cw310/README.md]: Done.

**Review by phil-levis — APPROVED** [review]

**Pull request merged**